internal/integration/emitter.go

```diff
@@ -74,10 +75,34 @@ func TelemetryHarvesterWithHarvestPeriod(t time.Duration) TelemetryHarvesterOpt
 
 // TelemetryHarvesterWithInfraTransport wraps the `telemetry.Harvester`
 // `Transport` so that it uses the `licenseKey` instead of the `apiKey`.
-func TelemetryHarvesterWithInfraTransport(licenseKey string) TelemetryHarvesterOpt {
-	return func(cfg *telemetry.Config) {
-		cfg.Client.Transport = newInfraTransport(cfg.Client.Transport, licenseKey)
+func TelemetryHarvesterWithInfraTransport(
```

Comment:
Is there a way we can have 2 separate functions here, for example:
`TelemetryHarvesterWithLicenseKey` and `TelemetryHarvesterWithProxy`, and chain both together to have one http.RoundTripper? I'm not a fan of mixing both the licenseKey & proxy support into one thing.

Reply:
done, I made the split between licenseKey, proxy and tlsConfig